source/MapDetailPanel.cpp

```diff
@@ -182,13 +211,15 @@ bool MapDetailPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command
 		{
 			plan.insert(plan.begin(), next);
 			Select(next);
+			GeneratePlanetCards(*selectedSystem);
 		}
 	}
 	else if((key == SDLK_DELETE || key == SDLK_BACKSPACE) && player.HasTravelPlan())
 	{
 		vector<const System *> &plan = player.TravelPlan();
 		plan.erase(plan.begin());
 		Select(plan.empty() ? player.GetSystem() : plan.front());
+		GeneratePlanetCards(*selectedSystem);
```

Comment:
Did you forget to set the scroll to 0 here? Setting the scroll to 0 should be part of GeneratePlanetCards so it can't be forgotten.